L1Trigger/TrackTrigger/interface/TTStubAlgorithm_globalgeometry.h

```diff
@@ -0,0 +1,131 @@
+#include <map>
+
+template< typename T >
+class TTStubAlgorithm_globalgeometry : public TTStubAlgorithm< T >
+{
+  private:
+    /// Data members
+    double mCompatibilityScalingFactor;
+    double mIPWidth;
+
+  public:
+    /// Constructor
+    TTStubAlgorithm_globalgeometry( const TrackerGeometry* const theTrackerGeom, const TrackerTopology* const theTrackerTopo, 
+                                    double aCompatibilityScalingFactor,
+                                    double aIPWidth ) : TTStubAlgorithm< T >( theTrackerGeom, theTrackerTopo, __func__ )
+    {
+      mCompatibilityScalingFactor = aCompatibilityScalingFactor;
+      mIPWidth = aIPWidth;
+    }
+
+    /// Destructor
+    ~TTStubAlgorithm_globalgeometry(){}
+
+    /// Matching operations
+    void PatternHitCorrelation( bool &aConfirmation,
+                                int &aDisplacement,
+                                int &anOffset,
+                                const TTStub< T > &aTTStub ) const;
+
+}; /// Close class
+
+/*! \brief   Implementation of methods
+ *  \details Here, in the header file, the methods which do not depend
+ *           on the specific type <T> that can fit the template.
+ *           Other methods, with type-specific features, are implemented
+ *           in the source file.
+ */
+
+/// Matching operations
+template< >
+void TTStubAlgorithm_globalgeometry< Ref_Phase2TrackerDigi_ >::PatternHitCorrelation( bool &aConfirmation,
+                                                                              int &aDisplacement,
+                                                                              int &anOffset,
+                                                                              const TTStub< Ref_Phase2TrackerDigi_ > &aTTStub ) const;
+
+/*! \class   ES_TTStubAlgorithm_globalgeometry
+ *  \brief   Class to declare the algorithm to the framework
+ *
+ *  \author Andrew W. Rose
+ *  \date   2013, Jul 18
+ *
+ */
+
+template< typename T >
+class  ES_TTStubAlgorithm_globalgeometry : public edm::ESProducer
+{
+  private:
+    /// Data members
+    boost::shared_ptr< TTStubAlgorithm< T > > _theAlgo;
+    double mPtThreshold;
+    double mIPWidth;
+
+  public:
+    /// Constructor
+    ES_TTStubAlgorithm_globalgeometry( const edm::ParameterSet & p ) :
+                                            mPtThreshold( p.getParameter< double >("minPtThreshold") ),
+                                            mIPWidth( p.getParameter< double >("ipWidth") )
+    {
+      setWhatProduced( this );
+    }
+
+    /// Destructor
+    virtual ~ES_TTStubAlgorithm_globalgeometry(){}
+
+    /// Implement the producer
+    boost::shared_ptr< TTStubAlgorithm< T > > produce( const TTStubAlgorithmRecord & record )
+    { 
+      /// Get magnetic field
+      edm::ESHandle< MagneticField > magnet;
+      record.getRecord< IdealMagneticFieldRecord >().get(magnet);
+      double mMagneticFieldStrength = magnet->inTesla( GlobalPoint(0,0,0) ).z();
```

Comment:
the code snipped above is used in many places in CMSSW, but I found the following alternative, that I think is equivalent:`double mMagneticFieldStrength = std::abs(magnet->nominalValue());`